### ISSUE TITLE
App Breaks On Interacting With Package

### ISSUE THREAD

**Issue by steveamaza** [wumke/react-native-immediate-phone-call#6]
Hello,

Can you please point me to what I am doing wrong? I recently made the decision to switch from `react-native-phone-call` to this package so numbers can be dialed directly from app without going to dialer but app breaks with no additional error message.

Here is a sample of my code;

```
...
import RNImmediatePhoneCall from 'react-native-immediate-phone-call';

testCall = () => {
      RNImmediatePhoneCall.immediatePhoneCall("0123456789");
    }

render() {
      return (
          <Button
            large={true}
            backgroundColor="#fec33a"
            color="#000000"
            onPress={this.testCall}
            title="Submit"
          />
      );
    }
  }
```
Here is the response that is returned from my logs

`11-08 18:17:34.941 27667 27667 D ReactNative: ReactInstanceManager.detachViewFromInstance()`

**Comment by wumke** [comment]
I cannot see immediately what's wrong either... on which platform and which version are you having this issue?
On higher android versions you must grant the app the permission to make phone calls, but you should see that in your logs. 
If you just console-log something inside the testCall function, does that work correctly?

**Comment by steveamaza** [comment]
Thanks for the quick response @wumke. I have granted permissions in my Manifest file but this persists.

Unfortunately, I have no idea what to console.log to see an error. Do you have any ideas?

**Comment by wumke** [comment]
You can try 
```
console.log('before');
RNImmediatePhoneCall.immediatePhoneCall("0123456789");
console.log('after');
```
Then you can see what happens in between, it seems highly unlikely that there would be no other log entry, except maybe if react native itself hangs...

Do you get the same behaviour when you make the call at view initialisation?

Which Android (test device os and build sdk) and react native version do you use?

**Comment by steveamaza** [comment]
Thanks.

I tried logging with a 'before' and 'after' statement, and it crashed right after running the last console.log.

```I/ReactNativeJS( 6871): before
I/ReactNativeJS( 6871): after
D/ReactNative( 6871): ReactInstanceManager.detachViewFromInstance()
```

However, after a few more tests, I figured what you were right by "I need to enable permissions". I thought you meant via my AndroidManifest.xml file (which I already did), but it seems that one needs to activate from the phone settings as well.

Maybe in a future update a prompt can be included to ask user to enable permission via phone Settings. I will probably send a PR when I get the chance
